test/expr/postfix/call/forward_trailing_closure_errors.swift

```diff
@@ -17,14 +17,98 @@ func testUnlabeledParamMatching(i: Int, fn: ((Int) -> Int) -> Void) {
   fn { $0 + i} // okay because the parameter label is empty
 }
 
-// When "fuzzy matching is disabled, this will fail.
-func forwardMatchFailure( // expected-note{{declared here}}
+func forwardMatchFailure(
   onError: ((any Error) -> Void)? = nil,
   onCompletion: (Int) -> Void
 ) { }
 
 func testForwardMatchFailure() {
   forwardMatchFailure { x in
     print(x)
-  } // expected-error{{missing argument for parameter 'onCompletion' in call}}{{4-4= onCompletion: <#(Int) -> Void#>}}
```

Comment:
I believe this example is actually expected to compile successfully in Swift 6 mode based on the behavior described in SE-0286